### DISCUSSION TITLE
Is there a way to do a partial read of a stream without qpdf buffering it all?

### DISCUSSION THREAD

**Discussion by jbarlow83** [qpdf/qpdf#948] [Q&A]
I'm trying to improve pikepdf's repr() implementation, and showing little snippets of streams makes sense to me (e.g. this looks like a content stream). It looks to me like all of the QPDF stream extraction APIs decode and buffer the stream in its entirely. Is there a way to do a partial extraction?

**Comment by jberkenbilt** [comment]
At this time, there is no way to do a partial extraction. The stream decoding is pretty deep, and with the need to apply a pipeline of filters potentially including decryption, implementing partial extraction meaningfully would be tricky. I suppose it might be possible to add something that would allow you to peek at the beginning of the stream by providing some mechanism for pipeStreamData to be told to stop sending the data when you've got enough back or maybe having a way to tell it to stop after a certain number of bytes.
